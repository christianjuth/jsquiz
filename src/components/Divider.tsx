import React from 'react';
import Theme from './Theme';

export function Divider({
  className
}: {
  className?: string
}) {
  const classes = Theme.useStyleCreatorClassNames(styleCreator);
  return (
    <div className={[className, classes.divider].join(' ')}/>
  );
}

function Vertical({
  className
}: {
  className: string
}) {
  const classes = Theme.useStyleCreatorClassNames(styleCreator);
  return (
    <div className={[className, classes.dividerVertical].join(' ')}/>
  );
}

const styleCreator = Theme.makeStyleCreator(theme => ({
  divider: {
    height: 1,
    width: '100%',
    backgroundColor: theme.colors.divider
  },
  dividerVertical: {
    width: 1,
    height: '100%',
    backgroundColor: theme.colors.divider
  }
}));

Divider.Vertical = Vertical;
export default Divider;