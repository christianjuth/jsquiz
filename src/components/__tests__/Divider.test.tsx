import React from 'react';
import renderer from 'react-test-renderer';
import Divider from '../Divider';

describe('<Index />', () => {

  test('renders correctly', () => {
    const tree = renderer.create(<Divider/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});