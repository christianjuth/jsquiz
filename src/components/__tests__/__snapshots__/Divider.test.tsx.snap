// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index /> renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#EEEEEE",
      "height": 1,
      "width": "100%",
    }
  }
/>
`;
